internal/server/v1/modules/mappers.go

```diff
@@ -0,0 +1,63 @@
+package modules
+
+import (
+	"encoding/json"
+
+	entropyv1beta1 "go.buf.build/odpf/gwv/odpf/proton/odpf/entropy/v1beta1"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/odpf/entropy/core/module"
+	"github.com/odpf/entropy/pkg/errors"
+)
+
+func moduleToProto(mod module.Module) (*entropyv1beta1.Module, error) {
+	var conf *structpb.Value
+	if len(mod.Configs) > 0 {
+		conf = &structpb.Value{}
+		if err := json.Unmarshal(mod.Configs, &conf); err != nil {
+			return nil, err
+		}
+	}
+
+	return &entropyv1beta1.Module{
+		Urn:       mod.URN,
+		Name:      mod.Name,
+		Configs:   conf,
+		Project:   mod.Project,
+		CreatedAt: timestamppb.New(mod.CreatedAt),
+		UpdatedAt: timestamppb.New(mod.UpdatedAt),
+	}, nil
+}
+
+func moduleFromProto(res *entropyv1beta1.Module) (*module.Module, error) {
+	confJSON, err := getConfigsAsRawJSON(res)
+	if err != nil {
+		return nil, err
+	}
+
+	return &module.Module{
+		URN:       res.GetUrn(),
+		Name:      res.GetName(),
+		Configs:   confJSON,
+		Project:   res.GetProject(),
+		CreatedAt: res.GetCreatedAt().AsTime(),
+		UpdatedAt: res.GetUpdatedAt().AsTime(),
+	}, nil
+}
+
+func getConfigsAsRawJSON(v interface{ GetConfigs() *structpb.Value }) ([]byte, error) {
+	errInvalidJSON := errors.ErrInvalid.WithMsgf("'configs' field must be specified and must be valid JSON")
+
+	var confJSON []byte
+	if confVal := v.GetConfigs(); confVal != nil {
+		var err error
+		confJSON, err = confVal.MarshalJSON()
+		if err != nil {
+			return nil, errInvalidJSON.WithCausef(err.Error())
+		}
+	} else {
+		return nil, errInvalidJSON
+	}
+	return confJSON, nil
```

Comment:
How about doing it like this? Could this be more readable?

```suggestion
	if confVal := v.GetConfigs(); confVal == nil {
		return nil, errInvalidJSON
	}
	confJSON, err := confVal.MarshalJSON()
	if err != nil {
		return nil, errInvalidJSON.WithCausef(err.Error())
	}
	return confJSON, nil
```

Reply:
can't do that exactly. `confVal` would not be accessible for MarshalJSON in this setup (confVal is scoped to the `confVal==nil` block only). but i can split it and clean it up a bit.